x/wasm/client/testutil/query.go

```diff
@@ -0,0 +1,466 @@
+
+	codeID, err := strconv.ParseUint(s.codeID, 10, 64)
+	s.Require().NoError(err)
+
+	testCases := map[string]struct {
+		args     []string
+		valid    bool
+		expected proto.Message
+	}{
+		"valid query": {
+			[]string{
+				s.contractAddress,
+			},
+			true,
+			&types.QueryContractInfoResponse{
+				Address: s.contractAddress,
+				ContractInfo: types.ContractInfo{
+					CodeID:    codeID,
+					Creator:   val.Address.String(),
+					Admin:     val.Address.String(),
+					Label:     "TestContract",
+					Created:   nil,
+					IBCPortID: "",
+					Extension: nil,
+				},
+			},
+		},
+		"no contractAddress": {
+			[]string{},
+			false,
+			nil,
+		},
+		"wrong contactAddress": {
+			[]string{
+				"abc",
+			},
+			false,
+			nil,
+		},
+	}
+
+	for name, tc := range testCases {
+		tc := tc
+
+		s.Run(name, func() {
+			cmd := cli.GetCmdGetContractInfo()
+			out, err := clitestutil.ExecTestCLICmd(val.ClientCtx, cmd, append(tc.args, s.queryCommonArgs()...))
+			if !tc.valid {
+				s.Require().Error(err)
+				return
+			}
+			s.Require().NoError(err)
+
+			var contractInfo types.QueryContractInfoResponse
+			s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &contractInfo), out.String())
+			s.Require().Equal(tc.expected, &contractInfo)
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestGetCmdGetContractStateAll() {
+	val := s.network.Validators[0]
+
+	testCases := map[string]struct {
+		args     []string
+		valid    bool
+		expected proto.Message
+	}{
+		"valid query": {
+			[]string{
+				s.contractAddress,
+			},
+			true,
+			&types.QueryAllContractStateResponse{
+				Models: []types.Model{
+					{
+						Key:   []byte("config"),
+						Value: []byte(fmt.Sprintf("{\"verifier\":\"%s\",\"beneficiary\":\"%s\",\"funder\":\"%s\"}", s.verifier, s.beneficiary, s.verifier)),
+					},
+				},
+				Pagination: &query.PageResponse{},
+			},
+		},
+		"wrong bech32_address": {
+			[]string{
+				"xxx",
+			},
+			false,
+			nil,
+		},
+		"no exist bech32_address": {
+			[]string{
+				"link1hmayw7vv0p3gzeh3jzwmw9xj8fy8a3kmpqgjrysljdnecqkps02qrq5rvm",
+			},
+			false,
+			nil,
+		},
+	}
+
+	for name, tc := range testCases {
+		tc := tc
+
+		s.Run(name, func() {
+			cmd := cli.GetCmdGetContractStateAll()
+			out, err := clitestutil.ExecTestCLICmd(val.ClientCtx, cmd, append(tc.args, s.queryCommonArgs()...))
+			if !tc.valid {
+				s.Require().Error(err)
+				return
+			}
+			s.Require().NoError(err)
+
+			var contractInfo types.QueryAllContractStateResponse
+			s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &contractInfo), out.String())
+			s.Require().Equal(tc.expected, &contractInfo)
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestGetCmdGetContractStateRaw() {
+	val := s.network.Validators[0]
+
+	testCases := map[string]struct {
+		args     []string
+		valid    bool
+		expected proto.Message
+	}{
+		"valid query": {
+			[]string{
+				s.contractAddress,
+				"636F6E666967", // hex.EncodeToString([]byte("config")),
```

Comment:
You may define a variable for it, instead of stating the logic in the comment.

Reply:
I wanted to use clear expected data, but I wanted to show everybody how this data is created. So I used this way.

Reply:
Then, how about adding lines like:
```
foo := hex.EncodeToString([]byte("config"))
s.Require().Equal("636F6E666967", foo)
```
and use the variable?
You can ignore my suggestion.

Reply:
I missed your update. I think it's fine as is. Ignore the above.